cartridges/int_adyen/cartridge/scripts/adyenRedirectVerificationSHA256.ds

```diff
@@ -129,45 +133,141 @@ function verify (args) {
 	adyenRequest.put("shopperReference", currentUser.ID );
 	adyenRequest.put("recurringContract", recurringType );
 	adyenRequest.put("resURL", URLUtils.https('Adyen-ShowConfirmation'));
-	
+	adyenRequest.put("countryCode", order.getBillingAddress().countryCode.toString().toUpperCase() );
+	// Optional Fields
+	//adyenRequest.put("shopperLocale", dw.system.Site.getCurrent().defaultLocale);
+
 	if(Site.getCurrent().getCustomPreferenceValue("Adyen_directoryLookup")){
 		if( !empty(args.brandCode)){
 			adyenRequest.put("brandCode", args.brandCode );
-		}		
+			if (args.brandCode == "ratepay" && !empty(args.ratePayFingerprint)){
+				adyenRequest.put("dfValue", args.ratePayFingerprint);
+			}else if (!empty(args.adyenFingerprint)){
+				adyenRequest.put("dfValue", args.adyenFingerprint);
+			}
+			if (args.brandCode == "klarna"){
+				adyenRequest.put("klarna.acceptPrivacyPolicy", "true");
+			}
+		}
 		if( !empty(args.issuerId)){
-			adyenRequest.put("issuerId", args.issuerId );	
+			adyenRequest.put("issuerId", args.issuerId );
 		}
 	}
-	
+
+	//Open Invoice specific section
+	//requires Gender and Date Of Birth (DOB) from Open Invoice form on the Billing Page and Phone Number from Billing Form
+	if(!empty(args.brandCode) && Site.getCurrent().getCustomPreferenceValue("Adyen_Open_Invoice_Whitelist").indexOf(args.brandCode) > -1) {
+		
+		var billingAddr = order.getBillingAddress();	
+		adyenRequest.put("shopper.firstName", order.getBillingAddress().getFirstName() );
+		adyenRequest.put("shopper.lastName", order.getBillingAddress().getLastName() );
+		adyenRequest.put("shopper.telephoneNumber", order.getDefaultShipment().getShippingAddress().getPhone() );
+		adyenRequest.put("shopperType", '1' );
+		adyenRequest.put("shopper.gender", (!empty(args.gender)) ? args.gender : 'NA');
+
+		if( !empty(args.dob)) {
+			var dob = args.dob.split('/');
+			adyenRequest.put("shopper.dateOfBirthDayOfMonth", dob[1] );
```

Comment:
This will be used mainly in EU, so we can either give dd/mm/yyyy as default format or ask for the unix timestamp